### PR TITLE
Hot Reload the OpenApi Document

### PR DESCRIPTION
**Pull request by aaronburtle**
## Why make this change?

Part of the ongoing work for https://github.com/Azure/data-api-builder/issues/67

this closes https://github.com/Azure/data-api-builder/issues/1869

This change will reload the open api docs during a hot reload scenario.

## What is this change?

Update the documentor class to get a new runtimeconfig rather than storing one internally. Then add a documentor to the runtimconfig provider, and pass it into the config file watcher. We then can call the create document function when we are hot reloading.

## How was this tested?

Testing is still needed. Will keep as draft until tested.

## Sample Request(s)

- Example REST and/or GraphQL request to demonstrate modifications
- Example of CLI usage to demonstrate modifications

### CODE REVIEW

**Comment by ayush3797** [comment]
This will also hot reload only the runtimeconfig section in the openapi doc? If rest.enabled changes from true to false, will the openapi doc no longer be available?

File: src/Core/Configurations/ConfigFileWatcher.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation.
```
Comment (abhishekkumams) [inline src/Core/Configurations/ConfigFileWatcher.cs:1]: why is this class being shown as newly added, this was already a part of your last PR that got merged.

File: src/Service/Startup.cs [hunk 2]
```diff
@@ -79,10 +79,10 @@ public void ConfigureServices(IServiceCollection services)
                 null);
             IFileSystem fileSystem = new FileSystem();
             FileSystemRuntimeConfigLoader configLoader = new(fileSystem, configFileName, connectionString);
+            services.AddSingleton<IOpenApiDocumentor, OpenApiDocumentor>();
```
Comment (abhishekkumams) [inline src/Service/Startup.cs:82]: Add a check to make sure we do this only when Rest is enabled. ref:  https://github.com/Azure/data-api-builder/pull/1872/files

File: src/Service.Tests/Configuration/ConfigurationTests.cs [hunk 3]
```diff
@@ -1337,6 +1337,67 @@ public async Task TestRuntimeBaseRouteInNextLinkForPaginatedRestResponse()
...
+
+                // we write to the config which is being monitored and sleep to allow hot reload to occur
+                fileSystem.File.WriteAllText(configFilePath, hotReloadConfig.ToJson());
+                Thread.Sleep(1000);
+                requestPath += "/MappedBookmarks";
+                request = new(httpMethod, requestPath);
+                response = await client.SendAsync(request);
+                Assert.IsTrue(response.StatusCode is HttpStatusCode.OK);
+                // overwrites the changes we made when hot reloading
+                fileSystem.File.WriteAllText(configFilePath, config.ToJson());
```
Comment (abhishekkumams) [inline src/Service.Tests/Configuration/ConfigurationTests.cs:1397]: what is the use of this last line in testing?

File: src/Core/Services/OpenAPI/OpenApiDocumentor.cs [hunk 4]
```diff
@@ -166,8 +167,9 @@ public void CreateDocument()
         private OpenApiPaths BuildPaths()
         {
             OpenApiPaths pathsCollection = new();
+            RuntimeConfig config = _configProvider.GetConfig();
```
Comment (abhishekkumams) [inline src/Core/Services/OpenAPI/OpenApiDocumentor.cs:170]: should we pass the config here from `CreateDocument()` ? is there a possibility the config that is sourced in `CreateDocument()` is different than the config sourced in `BuildPaths()`?

**Comment by seantleonard** [comment]
closing since this is really old. Please reopen a new PR with conflicts resolved and latest codebase changes included.

**Pull request closed without merging**